repo2docker_wholetale/matlab.py

```diff
@@ -0,0 +1,198 @@
+
+    This sets up MATLAB, JupyterLab and the
+    [matlab-kernel](https://github.com/Calysto/matlab_kernel
+    for a Tale with the following settings:
+
+    * `environment.json` contains `buildpack=MatlabBuildPack`.
+    * `toolboxes.txt` containing a valid product string for the
+       selected version.  See `installer_input.txt` from the Matlab
+       install media.
+
+    Prerequisites:
+    * Docker with buildkit support
+    * `repo2docker` with the `repo2docker_wholetale` plugin
+    * `matlab-install:<version>` image build using the
+      [matlab-install](https://github.com/whole-tale/matlab-install
+
+    For example:
+    repo2docker --user-name jovyan --user-id 1000 --no-run \
+            --build-arg FILE_INSTALLATION_KEY=<your key> \
+            --config ./repo2docker_config.py ./path-to-repo
+    """
+
+    _wt_env = None
+
+    def detect(self):
+        if not os.path.exists(self.binder_path("environment.json")):
+            return False
+
+        with open(self.binder_path("environment.json"), "r") as fp:
+            env = json.load(fp)
+
+        try:
+            return env["config"]["buildpack"] == "MatlabBuildPack"
+        except (KeyError, TypeError):
+            return False
+
+    def get_build_env(self):
+        # MLM_LICENSE_FILE specifies the path to the license at runtime
+        return super().get_build_env() + [("MLM_LICENSE_FILE", "/licenses/matlab/network.lic")]
+
+    def get_path(self):
+        """Adds path to Matlab binaries to user's PATH.
+        """
+        return super().get_path() + ["/usr/local/MATLAB/{}/bin".format(self.wt_env.get("VERSION", "R2020a"))]
+
+    def get_build_args(self):
+        """
+        Return args to be set at build time. FILE_INSTALLATION_KEY is
+        required only at build time.
+        """
+        return super().get_build_args() + ["FILE_INSTALLATION_KEY"]
+
+    def get_build_scripts(self):
+        """
+        Uses Docker builtkit to mount matlab install image to
+        install MATLAB core product. Installs Python engine and
+        Jupyter kernel.
+        """
+        return super().get_build_scripts() + [
+            (
+                "root",
+                r"""
+                wget -q https://xpra.org/gpg.asc -O- | apt-key add - && \
+                add-apt-repository "deb https://xpra.org/beta bionic main" && \
+                DEBIAN_FRONTEND=noninteractive apt-get install -y  xpra xpra-html5
+                """
+            ),
+            (
+                "root",
+                r"""--mount=type=bind,target=/matlab-install,source=/matlab-install/,from=matlab-install:{matlab_version} cd /matlab-install &&  ./install -mode silent -licensePath /matlab-install/network.lic -agreeToLicense yes -fileInstallationKey ${{FILE_INSTALLATION_KEY}} -product.MATLAB | grep --line-buffered -v fileInstallationKey | tee /dev/stderr | grep 'End - Successful' """.format(matlab_version=self.wt_env.get("VERSION", "R2020a"))
```

Comment:
Can you make this and similar lines more readable by splitting them? Something like:
```
(
    "long line but less than 120 chars"
    " && some other stuf..."
    " && moar!!! "
).format()
```

`f-strings` are also nice option instead of format, but totally optional.